src/middleware/AdaptiveCardSubmitActionHandler.ts

```diff
@@ -0,0 +1,22 @@
+import * as builder from "botbuilder";
+
+// Set the text field on message events to "adaptive card", if request is from an adaptive card
+export class AdaptiveCardSubmitActionHandler implements builder.IMiddlewareMap {
+
+    public readonly receive = (event: builder.IEvent, next: Function): void => {
+        if (event.type === "message")
+        {
+            let currEvent = (event as builder.IMessage);
+
+            // if event text is blank, replyToId is not null, event value is has isFromAdaptiveCard and messageText (please find these two fields in \src\dialogs\examples\basic\AdaptiveCardDialog.ts,
+            // submit action data field) in incoming payload to check if incoming request is from an adaptive card then set event text to messageText "adaptive card" to trigger Adaptive Card dialog,
+            // it's a work around that will be cleaned up later
```

Comment:
Let's remove the "it's a work around that will be cleaned up later" because this may stay way viable way of doing this.

In the comment above this, there is a typo with "is has". Let's have the comment be something like "Check X, Y, Z, etc to check if the incoming payload is from a Submit action button click from an AdaptiveCard (this is set in …/pathToDialog in the Submit action data field). If so, then set the text field of the incoming payload so the BotFramework regex recognizers will route the message to the desired dialog."


Hopefully that holds the information of what we are checking for, why we are checking it, and what/why we are taking the action we are in the if block.